js/screens/IngressScreen.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import React, { Component } from 'react'
+import { connect, type MapStateToProps } from 'react-redux'
+import { StyleSheet, ActivityIndicator } from 'react-native'
+import { NavigationActions, type NavigationScreenProp } from 'react-navigation'
+
+import { Container } from 'native-base'
+
+import ROUTES from '../navigation/routes'
+import type { Dispatch, AnyAction } from '../actions/types'
+import type { UserState } from '../reducers/user'
+
+type Props = {
+  user: UserState,
+  navigation: NavigationScreenProp<*, AnyAction>,
+  dispatch: Dispatch
+}
+
+/**
+ * An ingress screen to choose the real first screen the user must navigate to.
+ */
+class IngressScreen extends Component<Props> {
+  componentWillMount() {
+    const { user, dispatch } = this.props
+    if (!user.isLoggedIn) {
+      dispatch(this.navigate(ROUTES.LOGIN))
+    } else {
+      dispatch(this.navigate(ROUTES.HOME))
+    }
+  }
+
+  render() {
+    return (
+      <Container style={styles.container}>
+        <ActivityIndicator color={'blue'} />
+      </Container>
+    )
+  }
+
+  navigate = (route: string) => {
+    return NavigationActions.reset({
+      index: 0,
+      key: null,
+      actions: [
+        NavigationActions.navigate({
+          routeName: route
+        })
+      ]
+    })
+  }
+}
+
+const mapStateToProps: MapStateToProps<*, *, *> = (state: Object) => ({
```

Comment:
types of `MapStateToProps<S, OP, SP>` must be specified (don't use `*` nor `Object`)

Reply:
ditto

Reply:
I am going to fix all the typing issues this in another PR that's quite finished.